mne/encoding/model.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+from sklearn.base import BaseEstimator
+from scipy import linalg
+from ..externals.six import string_types
+
+
+class EncodingModel(object):
+    def __init__(self, est=None, delays=[0.], preproc_x=None, preproc_y=None):
+        """Fit a STRF model.
+
+        Fit an encoding model using time lags and a custom estimator.
+        Compatible either with continuous or events-based input
+        stimuli. It creates time lags for the input matrix, performs
+        preprocessing according to any transformers in the pipeline in `est`,
+        and then fits a model with the final estimator in `est`.
+
+        Parameters
+        ----------
+        est : instance of sklearn-style estimator
+            The estimator to use for fitting. This is any object that contains
+            a `fit` and `predict` method, which takes inputs of the form
+            (X, y), and which creates a `.coef_` attribute upon fitting.
+        delays : array, shape (n_delays,)
```

Comment:
What if we included tmin / tmax as optional parameters that could be used instead of delays? So if the user gives tmin/tmax, then it will do delaying in the way you've implemented. If the user gives `delays` then it'll explicitly create those delays.

In your implementation, how do you create delays that aren't continuous with one another (e.g., delays of 20, 40, 60 ms) without decimating data?